src/components/graph/graph.renderer.jsx

```diff
@@ -141,6 +141,53 @@ function _renderDefs() {
  */
 const _memoizedRenderDefs = _renderDefs();
 
+/**
+ * Generates the pattern defs for drawing the grid lines.
+ * This pattern is rendered inside of a rect in the graph.
+ *
+ * @param {Object} config - an object containing rd3g consumer defined configurations {@link #config config} for the graph.
+ * @param {Object} transform - object that represents current graph transform
+ * @returns Pattern defs for gridlines. Returns empty object if gridlines are disabled.
+ */
+const _gridLineDefs = (config, transform) => {
+  const { renderGridLines, spacingX, spacingY, color, smallStrokeSize, largeStrokeSize } = config.grid;
+
+  // grid lines turned off; return nothing
+  if (!renderGridLines) {
+    return {};
+  }
+
+  const width = spacingX * transform.k;
+  const height = spacingY * transform.k;
+
+  return (
+    <>
+      {/* This is the "inner" grid lines that are thinner than the outer ones */}
+      <pattern
```

Comment:
Code taken from Stack Overflow: https://stackoverflow.com/a/14209704/1006025

I had originally tried to do this with `line`s but it was not performant. With `pattern` like this, it works very nicely 👍

I had also tried doing this with [`d3-axis`](https://github.com/d3/d3-axis) but that did _not_ work out.